WordPress/gutenberg
# [RNMobile] [Embed block] Generic “No Preview” Embed preview UI

`gutenberg-mobile` PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/3927




## Description

Implements the Generic **No Preview** Embed preview UI when an embed is successful, but we're unable to show an inline preview.

## How has this been tested?
⚠️ I have not been able to find an Embed that recreates the scenario described below. I will continue my investigation to find a suitable candidate for testing. 

I thought any Facebook URL would work since its embed definitions have `previewable` as `false`. However, I wasn't able to get `false` when the embed was fetched.

Therefore I had to set `previewable` to `false` programmatically for the UI to be rendered via the logic below. 

https://github.com/WordPress/gutenberg/blob/f155d2744192fbdf958f0ce074956664b60e92aa/packages/block-library/src/embed/embed-no-preview.native.js#L206

1. Add an Embed block. 
2. Embed an URL that can be embedded but has no preview. 
3. Notice that the "No preview available" UI is displayed. 





## Screenshots 

<kbd><img src="https://user-images.githubusercontent.com/1509205/132414087-4a40db25-a107-4c2a-95b1-66d42a755d16.png" width="320"></kbd>

## Types of changes





The main changes that were implemented here are the `NoEmbedPreview` component was consolidated to support both the Provider **Coming Soon UI** and the Generic **No Preview UI**.

1. `isProviderPreviewable` ( false), `previewable` ( true) - this should show the Provider **Coming soon UI** if the previewable is set to true but the provider doesn't allow previews. Therefore the bottom sheet should open. 

2.  `isProviderPreviewable` ( true), `previewable` ( false) - This should show the Generic **No Preview UI**, since the content cannot be previewed at all. 

## Checklist:
